pkg/api/v1alpha1/cluster_types.go

```diff
@@ -148,14 +155,75 @@ type RegistryMirrorConfiguration struct {
 	InsecureSkipVerify bool `json:"insecureSkipVerify,omitempty"`
 }
 
+// OCINamespace represents an entity in a local reigstry to group related images.
+type OCINamespace struct {
+	// Name refers to the name of the upstream registry
+	Registry string `json:"registry,omitempty"`
+	// Namespace refers to the name of a namespace in the local registry
+	Namespace string `json:"namespace,omitempty"`
+}
+
 func (n *RegistryMirrorConfiguration) Equal(o *RegistryMirrorConfiguration) bool {
 	if n == o {
 		return true
 	}
 	if n == nil || o == nil {
 		return false
 	}
-	return n.Endpoint == o.Endpoint && n.Port == o.Port && n.CACertContent == o.CACertContent && n.InsecureSkipVerify == o.InsecureSkipVerify && n.Authenticate == o.Authenticate
+	return n.Endpoint == o.Endpoint && n.Port == o.Port && n.CACertContent == o.CACertContent && n.InsecureSkipVerify == o.InsecureSkipVerify && n.Authenticate == o.Authenticate && OCINamespacesSliceEqual(n.OCINamespaces, o.OCINamespaces)
+}
+
+// GetRegistryMirrorAddressMappings reads OCINamespaces field of RegistryMirrorConfiguration, and returns
+// a mapping from original registry URLs to registry mirror URLs.
+func (n *RegistryMirrorConfiguration) GetRegistryMirrorAddressMappings() map[string]string {
+	registryMirrorMappings := make(map[string]string)
+	base := net.JoinHostPort(n.Endpoint, n.Port)
+	// add registry mirror base address
+	registryMirrorMappings[constants.DefaultRegistryMirrorKey] = base
+	re := regexp.MustCompile(constants.DefaultPackageRegistryRegex)
+	// for each namespace, add corresponding endpoint
+	for _, ociNamespace := range n.OCINamespaces {
+		registry := base + "/" + ociNamespace.Namespace
+		if re.MatchString(ociNamespace.Registry) {
+			// handle curated packages in all regions
+			registryMirrorMappings[constants.DefaultPackageRegistryRegex] = registry
+		} else {
+			registryMirrorMappings[ociNamespace.Registry] = registry
+		}
+	}
+	if _, ok := registryMirrorMappings[constants.DefaultRegistry]; !ok {
+		registryMirrorMappings[constants.DefaultRegistry] = base
+	}
+	if _, ok := registryMirrorMappings[constants.DefaultPackageRegistryRegex]; !ok {
+		registryMirrorMappings[constants.DefaultPackageRegistryRegex] = base
```

Comment:
same comment here, I think we should ask users to do this explicitly.

Reply:
see the reply above.